packages/opentelemetry-core/src/platform/node/environment.ts

```diff
@@ -29,6 +31,91 @@ export function getEnv(): Required<ENVIRONMENT> {
   return Object.assign({}, DEFAULT_ENVIRONMENT, processEnv);
 }
 
+/**
+ * Retrieves a number from an environment variable.
+ * - Trims leading and trailing whitespace.
+ * - Returns `undefined` if the environment variable is empty, unset, contains only whitespace, or is not a number.
+ * - Returns a number in all other cases.
+ *
+ * @param {string} key - The name of the environment variable to retrieve.
+ * @returns {number | undefined} - The number value or `undefined`.
+ */
+export function getNumberFromEnv(key: string): number | undefined {
+  const raw = process.env[key];
+  if (raw == null || raw.trim() === '') {
+    return undefined;
+  }
+
+  const value = Number(raw);
+  if (isNaN(value)) {
+    diag.warn(
+      `Unknown value ${inspect(raw)} for ${key}, expected a number, using defaults`
+    );
+    return undefined;
+  }
+
+  return value;
+}
+
+/**
+ * Retrieves a string from an environment variable.
+ * - Returns `undefined` if the environment variable is empty, unset, or contains only whitespace.
+ *
+ * @param {string} key - The name of the environment variable to retrieve.
+ * @returns {string | undefined} - The string value or `undefined`.
+ */
+export function getStringFromEnv(key: string): string | undefined {
+  const raw = process.env[key];
+  if (raw == null || raw.trim() === '') {
+    return undefined;
+  }
+  return raw;
+}
+
+/**
+ * Retrieves a boolean value from an environment variable.
+ * - Trims leading and trailing whitespace and ignores casing.
+ * - Returns `undefined` if the environment variable is empty, unset, or contains only whitespace.
+ * - Returns `undefined` for strings that cannot be mapped to a boolean.
+ *
+ * @param {string} key - The name of the environment variable to retrieve.
+ * @returns {boolean | undefined} - The boolean value or `undefined`.
+ */
+export function getBooleanFromEnv(key: string): boolean | undefined {
+  const raw = process.env[key]?.trim().toLowerCase();
+  if (raw == null || raw === '') {
+    return undefined;
+  }
+  if (raw === 'true') {
+    return true;
+  } else if (raw === 'false') {
+    return false;
+  } else {
+    diag.warn(
+      `Unknown value ${inspect(raw)} for ${key}, expected 'true' or 'false', using defaults`
+    );
+    return undefined;
```

Comment:
Should these two `undefined` values be `false`?  The spec (https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#boolean) says:

> Any value not explicitly defined here as a true value, including unset and empty values, MUST be interpreted as false.

`undefined` is, of course, false-**y**, so not sure how strictly we want to take the spec. It might be slightly nice for downstream TypeScript users to have the tighter return type from this function.

Reply:
I've seen some places where it was counter-productive to deliver defaults from such an API in the past. Mostly with env-vars that have complex fallback structures where an unset env var has different semantics to the defaults - thinking about it more though, I don't think there's any such way which can apply to booleans in the same way.

I'll change it to return false in these cases.